website/docs/plugin/framework/resources/plan-modification.mdx

```diff
@@ -113,6 +113,59 @@ func UseStateForUnknown() planmodifier.Bool {
 }
 ```
 
+### Caveats
+
+#### Terraform Data Consistency Rules
+
+Terraform core [implements data consistency rules](https://github.com/hashicorp/terraform/blob/main/docs/resource-instance-change-lifecycle.md) between configuration, plan, and state data. For example, if an attribute value is configured, it is never valid to change that value in the plan except being set to null on resource destroy. The framework does not raise its own targeted errors in many situations, so it is the responsibility of the developer to account for these rules when implementing plan modification logic.
+
+#### Prior State Under Lists and Sets
```

Comment:
Would it be helpful to leave a link to this section of the docs on the actual Go pkg doc for the `UseStateForUnknown` functions?

https://pkg.go.dev/github.com/hashicorp/terraform-plugin-framework@v1.2.0/resource/schema/boolplanmodifier#UseStateForUnknown

Reply:
Oh shoot, I totally meant to put additional Go documentation for `UseStateForUnknown`. I'll push a commit with that.

Reply:
Does https://github.com/hashicorp/terraform-plugin-framework/pull/711/commits/60ae92bbfd90642ba50f7e7ea3f2c620d78137f3 look okay? I'm generally hesitant to link to website documentation unless its a dedicated page (its easy for section anchor links to change based adjustments to the Markdown header text).